pkg/vault/client.go

```diff
@@ -287,9 +287,28 @@ func (v Client) RevokeToken() error {
 // MustRevokeToken same as RevokeToken but the programm is terminated with an error if this fails.
 // Should be used in defer statements only.
 func (v Client) MustRevokeToken() {
-	if err := v.RevokeToken(); err != nil {
-		log.Entry().WithError(err).Fatal("Could not revoke token")
+
+	// only service tokens should be revoked and not batch tokens, the below will lookup the token and depends on the token prefix hvs. for service token
+	lookupPath := "auth/token/lookup-self"
+	const serviceTokenPrefix string = "hvs."
+	secret, err := v.GetSecret(lookupPath)
+
+	if err != nil {
+		log.Entry().Warnf("Could not lookup token at %s, not continuing to revoke", lookupPath)
+	} else {
+		if id, ok := secret.Data["id"]; ok {
+			if strings.HasPrefix(id.(string), serviceTokenPrefix) {
+				if err := v.RevokeToken(); err != nil {
+					log.Entry().WithError(err).Fatal("Could not revoke token")
+				}
+			} else {
+				log.Entry().Warnf("Service token not identified at %s, not continuing to revoke", lookupPath)
+			}
+		} else {
+			log.Entry().Warnf("Could not lookup token.Data.id at %s, not continuing to revoke", lookupPath)
+		}
```

Comment:
I'd remove the else statements and just return after each log, I think it makes it cleaner.

Reply:
I think it looks cleaner like:
```go
func (v Client) MustRevokeToken() {
	lookupPath := "auth/token/lookup-self"
	const serviceTokenPrefix = "hvs."

	secret, err := v.GetSecret(lookupPath)
	if err != nil {
		log.Entry().Warnf("Could not lookup token at %s, not continuing to revoke: %v", lookupPath, err)
		return
	}

	tokenID, ok := secret.Data["id"].(string)
	if !ok {
		log.Entry().Warnf("Could not lookup token.Data.id at %s, not continuing to revoke", lookupPath)
		return
	}

	if !strings.HasPrefix(tokenID, serviceTokenPrefix) {
		log.Entry().Warnf("Service token not identified at %s, not continuing to revoke", lookupPath)
		return
	}

	if err := v.RevokeToken(); err != nil {
		log.Entry().WithError(err).Fatal("Could not revoke token")
	}
}
```

Reply:
suggestion applied after resolving merge conflicts